lib/async_hooks.js

```diff
@@ -242,6 +242,12 @@ class AsyncResource {
                                   triggerAsyncId);
     }
 
+    // If type is null or undefined, set it by default
+    // to the new.target.name. This avoids an error when
+    // emitInitScript is called.
+    if (type == null)
```

Comment:
* This won't work with https://github.com/nodejs/node/pull/13610. /cc @refack 
* If the user types `new AsyncResource()` the name will conflict with another module.
* You can use `type = new.target.name` in the constructor.

Reply:
If we don't default then we should TypeError if the type isn't provided rather than the more obscure error that throws now.